src/utils.c

```diff
@@ -858,3 +859,28 @@ int gufi_dirfd(DIR *d) {
 
     return d_fd;
 }
+
+/*
+ * If a plugin library path is given in in->plugin_name,
+ * then load that library and add its plugin functions to in->plugin_ops.
+ *
+ * Returns 0 on success or 1 on failure.
+ */
+int load_plugin_library(struct input *in) {
+    if (!in->plugin_name.data) {
+        return 0;
+    }
+
+    void *lib = dlopen(in->plugin_name.data, RTLD_NOW);
+    if (!lib) {
+        fprintf(stderr, "Could not open plugin library: %s\n", dlerror());
+        return 1;
+    }
+
+    in->plugin_ops.db_init = dlsym(lib, "db_init");
+    in->plugin_ops.process_dir = dlsym(lib, "process_dir");
+    in->plugin_ops.process_file = dlsym(lib, "process_file");
+    in->plugin_ops.db_exit = dlsym(lib, "db_exit");
+
+    return 0;
+}
```

Comment:
I'd say move this to its own file or into bf.c since that's the only place it is used.